pkg/typesystem/typesystem.go

```diff
@@ -546,9 +546,10 @@ func validateRelationRewrites(model *openfgapb.AuthorizationModel) error {
 	return nil
 }
 
-// isUsersetRewriteValid checks if a particular userset rewrite is valid. The first argument is all the relations in
-// the typeSystem, the second argument is the subset of relations on the type where the rewrite occurs.
+// isUsersetRewriteValid checks if a particular userset rewrite is valid. The second argument is all the relations in
+// the typeSystem, the third argument is the subset of relations on the type where the rewrite occurs.
 func isUsersetRewriteValid(
+	typesys *TypeSystem,
 	allRelations map[string]*openfgapb.Relation,
```

Comment:
If we're going to take the typesystem then we should omit the `allRelations` part from the signature and the `relationsOnType`, because those should come included with helpers on the typesystem itself.

Reply:
Done. And simplifed!